DllImportGenerator/DllImportGenerator/Marshalling/StringMarshaller.Utf16.cs

```diff
@@ -0,0 +1,298 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.Interop
+{
+    internal class Utf16StringMarshaller : IMarshallingGenerator
+    {
+        private static readonly TypeSyntax InteropServicesMarshalType = ParseTypeName(TypeNames.System_Runtime_InteropServices_Marshal);
+        private static readonly TypeSyntax NativeType = PointerType(PredefinedType(Token(SyntaxKind.UShortKeyword)));
+        private static readonly int StackAllocBytesThreshold = 260 * sizeof(ushort);
```

Comment:
The "logic" for the 260 is based on the `MAX_PATH` variable in Windows. The assumption being that file path are the most used case for strings and therefore we should optimize for the most common length (i.e. 260). I would like to capture that logic so in the future we can make an informed decision about possibly changing it. Feel free to mark this as a `[Compat]`.